core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -432,8 +432,10 @@ private[spark] class ExecutorAllocationManager(
     if (testing || executorsRemoved.nonEmpty) {
       executorsRemoved.foreach { removedExecutorId =>
         newExecutorTotal -= 1
+        val hasCachedBlocks = SparkEnv.get.blockManager.master.hasCachedBlocks(executorId);
```

Comment:
This variable `executorId` is not defined, should be change to `removedExecutorId`.